runelite-client/src/main/java/net/runelite/client/plugins/theatreofblood/TheatreOfBloodPlugin.java

```diff
@@ -438,6 +494,128 @@ protected void onClientTick(ClientTick event)
 					break;
 				case NYLOCAS:
 					currentEncounter.castToNylocas().checkNylocasAggressiveNpcs(client.getNpcs(), client.getPlayers());
+					currentEncounter.castToNylocas().checkNylocasTimers(client.getGameCycle());
+					break;
+				case XARPUS:
+					if (currentEncounter.castToXarpus().getIsStaring())
+					{
+						currentEncounter.castToXarpus().checkTurnTimer(client.getGameCycle());
+					}
+					break;
+				case VERZIK_VITUR:
+					if (currentEncounter.castToVerzik().getVerzikPhase() != 0)
+					{
+						currentEncounter.castToVerzik().checkAttackTimer(client.getGameCycle());
+					}
+					break;
+			}
+		}
+	}
+
+	@Subscribe
+	protected void onVarbitChanged(VarbitChanged varbitChanged)
+	{
+		if (client.isInInstancedRegion() && currentEncounter != null && currentEncounter.getEncounter() != null)
+		{
+			switch (currentEncounter.getEncounter())
+			{
+				case SOTETSEG:
+					if (varbitChanged.getIndex() == 1745)
```

Comment:
1745 is a [magic number](https://en.wikipedia.org/wiki/Magic_number_(programming)) and should be given an constant value to explain what it is

Reply:
Note: this function was moved to the `Sotetseg` class

Reply:
Removed `getIndex()` check as it was unnecessary